e2e/provision/gce_init.sh

```diff
@@ -49,14 +47,18 @@ chown $NEPHIO_USER:$NEPHIO_USER /home/$NEPHIO_USER/.bash_aliases
 sed -e "s/vagrant/$NEPHIO_USER/" < /home/$NEPHIO_USER/test-infra/e2e/provision/nephio.yaml > /home/$NEPHIO_USER/nephio.yaml
 cd ./test-infra/e2e/provision
 export DEBUG DEPLOYMENT_TYPE
-runuser -u $NEPHIO_USER ./gce_install_sandbox.sh
+
+if ! getent group docker > /dev/null; then
+    addgroup docker
+fi
```

Comment:
Yes, but if you don't add it before calling the gce_install_sandbox.sh script using the nephio user, you can't add the nephio user to the docker group. In a shell, while you can add a user to a group, you can't execute anything using that new group's rights without spawning a new shell.

I know this is not a perfect way of doing this and we should restructure these scripts afterwards, but it does work and the Docker install works fine even if the docker user already exists.

Reply:
My understanding is that line 57 adds `$NEPHIO_USER` into the docker group, correct me if I'm wrong, but is this not doing the same?

BTW, I was planning to move 56-58 into the `gce_install_sandbox.sh` script.

Reply:
I just tried this:
sudo usermod -aG docker2 ubuntu
usermod: group 'docker2' does not exist

(used non-existant docker2 group as a test)
The group has to exist before it can be given to a user.

I tried moving lines 56-58 into the gce_install_sandbox.sh script but because of the requirement to restart the shell, docker or kpt fn render commands later in the script would not run as $NEPHIO_USER because a new shell has to be started to pick up the new group privileges

Reply:
We probably should have one phase in the install process to install all the prerequisites (docker kpt kubectl etc etc) and then in another phase install the more nephio-specific stuff we need but I guess that'll be post R1 :-)